### PR TITLE
Added Oracle implementation

### PR DESCRIPTION
**Pull request by anoordover**

### CODE REVIEW

**Comment by Giorgi** [comment]
Did you test it on a real Oracle database?

**Comment by anoordover** [comment]
Yes, I did on my own environment. The constants I use are part of the so called ORA messages e.g. [ORA-02291](http://www.dba-oracle.com/t_ora_02291_integrity_constraint_string_string_violated_parent_key_not_found.htm)

**Comment by Giorgi** [comment]
Yes, I'm aware of Ora errors. The reason I'm asking is that from my experience in some cases the library throws `OracleTruncateException` when there is numeric overflow.

**Comment by anoordover** [comment]
I think this Exception is only thrown by de non-core Oracle.ManagedDataAccess en the Oracle.DataAccess libraries

**Comment by anoordover** [comment]
I did an additional test using a decimal and I don't get this error. Is still get the ORA-01438 error (putting 10.001m in a number(5,4) column)

**Comment by anoordover** [comment]
I tried another integrationtest but I stil didn't get this exception. I don't know how to get this type of exception... Passing a value too large for a long in a long when executing a select-query.

**Comment by Giorgi** [comment]
Can you add those tests to the tests project so that I can run them locally? Preferably following the same structure and covering all cases as the current tests do.

**Comment by anoordover** [comment]
> Can you add those tests to the tests project so that I can run them locally? Preferably following the same structure and covering all cases as the current tests do.

I added unittests for the Oracle implementation.
`-- auto-generated definition
create table "Product"
(
    "Id"   NUMBER            not null
        constraint PRODUCT_PK
            primary key,
    "Name" VARCHAR2(15 char) not null
)
/

create unique index PRODUCT_NAME_UINDEX
    on "Product" ("Name")
/

-- auto-generated definition
create table "ProductSale"
(
    "Id"        NUMBER       not null
        constraint PRODUCTSALE_PK
            primary key
        constraint PRODUCTSALE_PRODUCT_ID_FK
            references "Product",
    "Price"     NUMBER(5, 2) not null,
    "SoldAt"    DATE,
    "ProductId" NUMBER       not null
)
/


`

**Comment by Giorgi** [comment]
I'll check it in a couple of days.

**Comment by anoordover** [comment]
I can't find how to skip all the test in a class in XUnit except by making the test-class non-public.
Do you know another way to skip all the tests of a whole class?

**Comment by Giorgi** [comment]
There is no such way in xUnit. The only way is to override every test method and mark it as skipped. 

Making the class private will remove those test from skipped section in unit tests results.

File: EntityFramework.Exceptions.Tests/OracleTests.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using Xunit;
+
+namespace EntityFramework.Exceptions.Tests
+{
+    public class OracleTests : DatabaseTests, IClassFixture<OracleTestsontextFixture>
+    {
+        public OracleTests(OracleTestsontextFixture fixture) : base(fixture.Context)
+        {
+        }
+
```
Comment (Giorgi) [inline EntityFramework.Exceptions.Tests/OracleTests.cs:13]: Please remove empty line

File: EntityFramework.Exceptions.Tests/OracleTests.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+    
+    public class OracleTestsontextFixture : DemoContextFixture
```
Comment (Giorgi) [inline EntityFramework.Exceptions.Tests/OracleTests.cs:16]: `Context` instead of `sontext`

File: EntityFramework.Exceptions.Tests/OracleTests.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        protected override DbContextOptionsBuilder<DemoContext> BuildOptions(DbContextOptionsBuilder<DemoContext> builder, IConfigurationRoot configuration)
+        {
+            //var connectionString = configuration.GetConnectionString("Oracle");
+            var connectionString = "Data Source=brio.WORLD;Persist Security Info=True;User ID=bri;Password=bri;";
```
Comment (Giorgi) [inline EntityFramework.Exceptions.Tests/OracleTests.cs:21]: Remove the hardcoded sting and uncomment the line above

**Comment by Giorgi** [comment]
I ran the tests and it worked fine. Can you make the changes according to the comments that I left? Also, add the attribute to the Oracle tests so they get skipped because AppVeyor doesn't support oracle 😞

**Comment by Giorgi** [comment]
@anoordover Do you think that it should handle Ora-02292 too?

**Comment by anoordover** [comment]
To be honest, I think you are right to say that we should support ORA-02292 as well.

**Pull request merged**